Sources/DittoPresenceDegradation/PresenceDegradationVM.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  File.swift
+//  
+//
+//  Created by Walker Erekson on 2/13/24.
+//
+
+import Foundation
+import DittoSwift
+import CryptoKit
+import Combine
+
+@available(iOS 15.0, *)
+class PresenceDegradationVM: ObservableObject {
+        
+    @Published var expectedPeers: Int = 0
+    @Published var apiEnabled: Bool = false
+    @Published var sessionStartTime: String?
+    @Published var isSheetPresented = true
+    @Published var localPeer: Peer?
+    @Published var remotePeers: [String:Peer]?
+    @Published var settings: Settings?
+    var ditto: Ditto
+    var peersObserver: DittoObserver?
+
+    init(ditto: Ditto) {
+        self.ditto = ditto
+    }
+    
+    func startNewSession() {
+                
+        if(peersObserver != nil) {
+            self.peersObserver?.stop()
+        }
```

Comment:
Minor - the optional chaining down inside means we actually don't need the nil check here, `stop()` won't be executed if `peersObserver` is `nil`